components/formats-bsd/src/loci/formats/in/FitsReader.java

```diff
@@ -137,19 +147,29 @@ protected void initFile(String id) throws FormatException, IOException {
       else if (key.equals("NAXIS2")) m.sizeY = Integer.parseInt(value);
       else if (key.equals("NAXIS3")) m.sizeZ = Integer.parseInt(value);
 
+      if (key.equals("NAXIS")) naxis = Integer.parseInt(value);
       addGlobalMeta(key, value);
     }
     while (in.read() == 0x20);
     pixelOffset = in.getFilePointer() - 1;
 
     m.sizeC = 1;
     m.sizeT = 1;
-    if (getSizeZ() == 0) m.sizeZ = 1;
+    if (getSizeZ() == 0) {
+        m.sizeZ = 1;
+    }
+    else m.sizeZ = getSizeZ();
+
+    // If naxis is 1, the other dimensions need to be set manually
+    if (naxis == 1){
+        m.sizeY = 1;
+        m.sizeZ = 1;
+    }
 
     // correct for truncated files
     int planeSize =
-      getSizeX() * getSizeY() * FormatTools.getBytesPerPixel(getPixelType());
-    if (DataTools.safeMultiply64(planeSize, getSizeZ()) >
+      m.sizeX * m.sizeY * FormatTools.getBytesPerPixel(getPixelType());
+    if (DataTools.safeMultiply64(planeSize, m.sizeZ) >
```

Comment:
```m.sizeX``` and ```getSizeX()``` should be equivalent (ditto for Y and Z) - was there a reason for this change?